src/librbd/AioObjectRequest.cc

```diff
@@ -547,8 +552,13 @@ void AbstractAioObjectWrite::send_write_op(bool write_guard)
 
   librados::AioCompletion *rados_completion =
     util::create_rados_safe_callback(this);
-  int r = m_ictx->data_ctx.aio_operate(m_oid, rados_completion, &m_write,
-                                       m_snap_seq, m_snaps);
+  int r;
+  if (m_trace && m_trace.valid())
```

Comment:
My point was that since ```m_trace``` isn't a pointer, it has a ```bool``` operator that just calls ```valid()```, therefore, it's redundant to test both here.